test/ctia/test_helpers/graphql.clj

```diff
@@ -29,3 +33,135 @@
                       :headers {"api_key" "45c1f5e3f05d0"})]
     (into body
           {:status status})))
+
+(defn nodes->edges
+  [nodes]
+  (mapv (fn [node]
+          {:node node})
+        nodes))
+
+(defn sort-kw->path
+  ":valid_date.start_time -> (:valid_date :start_time)"
+  [kw]
+  (map
+   keyword
+   (some-> kw
+           name
+           (str/split #"\."))))
+
+(defn min-value
+  "Takes the min value if the value is sequential.
+  Equivalent to the min Sort mode in ElasticSearch"
+  [v]
+  (if (sequential? v)
+    (first (sort v))
+    v))
+
+(defn connection-sort-test
+  "Test a connection with a list of sort-fields.
+  It uses the $sort_field variable to specify a sort field.
+  The specified query/operation-name should provide it."
+  [operation-name
+   graphql-query
+   variables
+   connection-path
+   sort-fields]
+  (doseq [sort-field sort-fields]
+    (let [{:keys [data errors status]}
+          (query graphql-query
+                 variables
+                 operation-name)
+          connection-data (get-in data connection-path)
+          sort-field-path (sort-kw->path sort-field)
+          sort-field-fn #(min-value
+                          (get-in % sort-field-path))
+          nodes-ref (->> (:nodes connection-data)
+                         (sort-by sort-field-fn)
+                         (map sort-field-fn)
+                         (remove nil?))
+          edges-ref (->> (:edges connection-data)
+                         (map :node)
+                         (sort-by sort-field-fn)
+                         (map sort-field-fn)
+                         (remove nil?))]
+      (is (= 200 status))
+      (is (empty? errors) "No errors")
+      (let [{:keys [data errors status]}
+            (query graphql-query
+                   (into variables
+                         {:sort_field (sorting/sorting-kw->enum-name sort-field)})
+                   operation-name)
+            connection-data (get-in data connection-path)
+            nodes (->> (:nodes connection-data)
+                       (map sort-field-fn)
```

Comment:
as sort-field-fn is used by the the graphql query. I think it would be nice to have one manual test which compare datas directly to check that sort is doing something.